theano/gof/opt.py

```diff
@@ -3039,3 +3039,34 @@ def check_stack_trace(f_or_fgraph, ops_to_check='last', bug_print='raise'):
                 return False
 
     return True
+
+
+class CheckStrackTraceFeature(object):
+    def on_import(self, fgraph, node, reason):
+        # In optdb we only register the CheckStackTraceOptimization when
+        # theano.config.check_stack_trace is not off but we also double check here.
+        if theano.config.check_stack_trace != 'off' and not check_stack_trace(fgraph, 'all'):
+            if theano.config.check_stack_trace == 'raise':
+                    raise NotImplementedError(
+                        'Empty stack trace! The optimization that inserted this variable is' + str(reason))
```

Comment:
Raising something else than `AssertionError` will, by default, just cause the optimization to be skipped (that is, unless `config.on_opt_error='raise'`).
So I would prefer to use `AssertionError` rather than `NotImplementedError` here.
Also don't forget a space between 'is' and 'reason'.